[WIP Tracker] Update to 1.19

Rover656 opened the pull request:
# Description

Slowly working on porting to 1.19. Will not build let alone run yet.
Waiting for us to be able to run on latest forge with registrate before merge.

Fixes #59 

# Todo

- [x] Buildable
- [x] Runnable
- [ ] Fix issues
- [ ] Tidy the new code
- [ ] Review old practice, make sure we're following standards.
- [x] Wait for PR's to be merged
- [x] Fix merge
- [x] Upgrade to latest forge
- [x] Wait for registrate
- [ ] Wait for registrate to update past fluid rework
- [x] New codesplitting technique

# Issues:

- [x] Energy and progress tooltips don't display numbers
- [ ] Fix door datagen [MinecraftForge/MinecraftForge#8687](https://github.com/MinecraftForge/MinecraftForge/pull/8687)
- [x] Check datagen is fixed in EIODecor once we bump past forge 41.0.41

**Note:** this PR got big, so some of the 1.19 porting stuff will continue in future PRs now. Just waiting for the Registrate PR then we'll merge.

# Checklist:

- [x] My code follows the style guidelines of this project (.editorconfig, most IDEs will use this for you)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation

<!-- Thanks to: https://embeddedartistry.com/blog/2017/08/04/a-github-pull-request-template-for-your-projects/ for the building blocks of this template -->

Rover656 committed:
Upgrade forgegradle

Rover656 committed:
First pass on registry changes.

Rover656 committed:
Fix rogue refactor

justliliandev commented on build.gradle:
```diff
@@ -104,16 +106,16 @@ allprojects {
         shade registrate
 
         // JEI
-        compileOnly fg.deobf("mezz.jei:jei-${minecraft_version}:${jei_version}:api")
-        runtimeOnly fg.deobf("mezz.jei:jei-${minecraft_version}:${jei_version}")
+//        compileOnly fg.deobf("mezz.jei:jei-${minecraft_version}:${jei_version}:api")
+//        runtimeOnly fg.deobf("mezz.jei:jei-${minecraft_version}:${jei_version}")
```
Those seem to be available

Rover656 replied:
They are now, but weren't when I started the port branch :)

justliliandev commented on enderio-base/src/main/java/com/enderio/base/EnderIO.java:
```diff
@@ -1,31 +1,21 @@
-import com.enderio.base.common.init.EIOItems;
-import com.enderio.base.datagen.loot.FireCraftingLootProvider;
-import com.enderio.base.datagen.tags.EIOBlockTagsProvider;
+import com.enderio.base.common.init.*;
```
is star import wanted?

Rover656 replied:
I don't think it hurts anything really, unless we really want to avoid wildcards. but to be fair all init classes are used for init.

justliliandev replied:
alright

Rover656 committed:
Rollback forge and fix remaining errors

justliliandev commented on enderio-base/src/main/java/com/enderio/base/common/item/LocationPrintoutItem.java:
```diff
@@ -74,7 +73,7 @@ private static void openMenu(ServerPlayer player, CoordinateSelection selection,
             @Override
             public Component getDisplayName() {
-                return new TextComponent("");
+                return Component.literal("");
```
Component.empty()

Rover656 committed:
Fix xp boost handler and some icon rendering

justliliandev commented on enderio-base/src/main/java/com/enderio/base/common/loot/CapacitorLootModifier.java:
```diff
@@ -47,7 +44,7 @@ protected CapacitorLootModifier(LootItemCondition[] conditionsIn, float min, flo
      */
     @Override
-    protected List<ItemStack> doApply(List<ItemStack> generatedLoot, LootContext context) {
+    protected @NotNull ObjectArrayList<ItemStack> doApply(ObjectArrayList<ItemStack> generatedLoot, LootContext context) {
```
everything is notnull by default

Rover656 committed:
Fixed a strange crash and a couple other things

Rover656 committed:
Install JEI v11

Rover656 committed:
Merge branch 'dev/1.18.x' into update/1.19

Rover656 committed:
Fix merge, official registrate, fix and run datagen

Rover656 committed:
Lots of tidying

ferriarnus committed:
Light and Powered Light blocks (#21)

* Light works somewhat

light updates need to be fixed for chunk borders
Need the actual "power" part
Need inverted and redstone logic

* Redstone logic

* Let there be light

Finally lights work (tm)
Some cleanup will be needed, especially the network stuff, since this will be joined in base.

* Update PoweredLightBlockEntity.java

oops

* first round of fixes

* first doc pass

* Update DecorBlockState.java

* Update PoweredLightBlockEntity.java

Rover656 committed:
Forcibly deleting a file with broken name

Rover656 committed:
Readd with correct name

Rover656 committed:
Apply Ferri's gradle changes

Rover656 committed:
Merge branch 'dev/1.18.x' into update/1.19

Rover656 committed:
rerun datagen

Rover656 committed:
Disable combined jar and fix shadowing properly

Rover656 committed:
For some reason datagen was misfiring.

Rover656 committed:
Fix datagen temporarily.

Rover656 committed:
Switch to jarjar

Rover656 committed:
Bump to latest forge and Registrate#45

Rover656 committed:
Fix and rerun data.

Rover656 committed:
More work, brace for large change count

Rover656 committed:
Complete project restructure

Rover656 committed:
Versioning and tidy the subproject logic

Rover656 committed:
Mixins, maven publishing and other jars

Rover656 committed:
Bump forge, fix machines datagen and use parchment mappings

Rover656 committed:
Some misc todos

justliliandev commented on build.gradle:
```diff
@@ -256,3 +233,80 @@ publishing {
+// Examples:
+//   * enderio-base-1.12.2-4.0.1.jar            :: release version 4.0.1 for mc 1.12.2
+//   * enderio-base-1.12.2-4.2.1-nightly-4      :: nightly build no. 4 for version 4.2.1
+//   * enderio-base-1.12.2-4.5.1-dev-c91c8ee6e  :: dev (local) build for commit c91c8ee6e
```
since we don't provide any modules this should all be enderio-\<mc-version\>, right?

Rover656 replied:
correct, i missed this when copying this block :P

Rover656 committed:
More core to its own source set.

Rover656 committed:
Fix versioning comments

Rover656 committed:
Running registrate off a temporary maven

Rover656 commented:
Gonna go for merging this soon, waiting for the build to hopefully pass first

Rover656 committed:
Missed another comment
